cli/internal/run/run.go

```diff
@@ -912,9 +904,16 @@ func (e *execContext) exec(pt *packageTask) error {
 		if err != nil {
 			targetUi.Error(fmt.Sprintf("error fetching from cache: %s", err))
 		} else if hit {
-			if e.rs.Opts.stream && fs.FileExists(filepath.Join(e.rs.Opts.cwd, logFileName)) {
-				e.logReplayWaitGroup.Add(1)
-				go replayLogs(targetLogger, e.ui, e.rs.Opts, logFileName, hash, &e.logReplayWaitGroup, false, e.rs.Opts.cacheHitLogsMode)
+			switch e.rs.Opts.cacheHitLogsMode {
+			case HashLogs:
+				targetUi.Output(fmt.Sprintf("cache hit, suppressing output %s", ui.Dim(hash)))
+			case FullLogs:
+				if e.rs.Opts.stream && fs.FileExists(filepath.Join(e.rs.Opts.cwd, logFileName)) {
+					e.logReplayWaitGroup.Add(1)
+					go replayLogs(targetLogger, e.ui, e.rs.Opts, logFileName, hash, &e.logReplayWaitGroup, false)
```

Comment:
I realize it was here earlier, but let's get rid of the `false` for `silent` too. I think this is the only place where `replayLogs` is called.

Reply:
Done!